File: Semester3/Advanced Methods of Programming/Toy Language Interpreter C#/Assignement7/Assignement7/Model/PrintStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignement7.Model
{
    public class PrintStatement : IStatement
    {
        Expression expression;

        public PrintStatement(Expression expression)
        {
            this.expression = expression;
        }


        public override string ToString()
        {
            return "print( " + expression.ToString() + " )";
        }


        public ProgramState Execute(ProgramState state)
        {
            state.getOutputList().Add(expression.Evaluate(state.getSymbolTable()));
            return state;
        }
    }
}

File: Semester3/Advanced Methods of Programming/ExamenMAP/src/model/statements/PrintStatement.java
package model.statements;

import model.ProgramState;
import model.exceptions.DivisionByZeroException;
import model.expressions.Expression;

public class PrintStatement implements IStatement{
    private Expression variableName;

    public PrintStatement(Expression variableName)
    {
        this.variableName = variableName;
    }

    @Override
    public ProgramState execute(ProgramState state) throws Exception {
        state.getOutputList().add(variableName.evaluate(state.getSymbolTable(), state.getHeapTable()));
        return null;
    }

    @Override
    public String toString(){
        return "print( " + variableName + " )";
    }
}
